src/Core/Domain/AttributeGroup/Command/EditAttributeGroupCommand.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupId;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupType;
+
+/**
+ * Edits existing attribute group
+ */
+class EditAttributeGroupCommand
+{
+    /**
+     * @var AttributeGroupId
+     */
+    private $attributeGroupId;
+
+    /**
+     * @var string[]
+     */
+    private $localizedNames;
+
+    /**
+     * @var int[]
+     */
+    private $shopAssociation;
+
+    /**
+     * @var array
+     */
+    private $localizedPublicNames;
+
+    /**
+     * @var AttributeGroupType
+     */
+    private $type;
+
+    /**
+     * @param string[] $localizedNames
+     * @param int[] $shopAssociation
+     *
+     * @throws AttributeGroupConstraintException
+     */
+    public function __construct(
+        int $attributeGroupId,
+        array $localizedNames,
+        array $localizedPublicNames,
+        AttributeGroupType $type,
+        array $shopAssociation = []
```

Comment:
```suggestion
        array $shopAssociation = null
```
The default value which is to not modify anything should be null (that's how we identify the partial update in our recent commands). And when this value is not null it should be a non empty array, empty shop association is basically the same as removing the entity itself so it shouldn't be allowed in an update command